File: others/src/main/scala/com/example/others/scala/TestScala.scala
package com.example.others.scala

import java.util

import org.apache.commons.collections.ComparatorUtils

class TestScala(private val c1: C1) {
    c1.c1Print()

}

class C1(val name: String, val id: Long) {

    def c1Print() {
        println(name)
        println(id)
    }
}

object TestScala {
    def main(args: Array[String]): Unit = {
////        val testScala = new TestScala(new C1("c1_1", 100L))
//        val list :java.util.List[String] =new util.ArrayList[String]()
//        list.add("sssss")
//        list.add(null)
//        val List(k1,null) = list
////        println(k1)
////        println(k2)
//        Array(1,2,3).mkString(",")

      val a = List(("a",10),("b",10),("a",11),("a",9))

      println(a.sorted)
      println(a.sortWith((i1,i2) => i1._1 < i2._1 || i1._1 ==i2._1 && i1._2<i2._2)  )
    }
}

File: others/src/main/java/com/example/others/java/TestScala.java
package com.example.others.java;

import java.util.ArrayList;
import java.util.List;

import scala.Function1;
import scala.collection.JavaConverters;
import scala.collection.Seq;

public class TestScala {
    public static void main(String[] args) {


        List<String> list = new ArrayList<>();
        list.add("a");
        list.add("a");
        list.add("a");
        list.add("b");
        Seq<String> seq = JavaConverters.asScalaIteratorConverter(list.iterator()).asScala().toSeq();




    }
}
